arch/risc-v/src/common/espressif/Bootloader.mk

```diff
@@ -1,5 +1,5 @@
 ############################################################################
-# arch/risc-v/src/espressif/Bootloader.mk
+# arch/risc-v/src/common/espressif/Bootloader.mk
```

Comment:
To divide the chip specific files into separate folders (e.g risc-v/src/esp32c6) for having chip specific support or config, `risc-v/src/espressif` folder has been moved. Also this approach avoids potential misunderstanding about actual src file path.

Reply:
> Also this approach avoids potential misunderstanding about actual src file path.

@xiaoxiang781216 , about `arch/risc-v/src/espressif/common`: `espressif` doesn't refer to the chip itself (unlikely, for instance, ARM's `nuttx/arch/arm/src/armv7-a` which is also common to other chips). It refers to a set of sources that are shared between different Espressif chips. So, in order to keep consistency, we created an  `espressif` folder in `arch/risc-v/src/common/espressif`. That was the idea...

Reply:
but armv7-a is at the same level of common, not put into common/armv7-a.

Reply:
> but armv7-a is at the same level of common, not put into common/armv7-a, so it's strange that riscv/common contain espressif.

Yes, that is the point I highlighted: `armv7-a` refers to the core of an arch that is being included by other chips. `espressif` does not...

The level `arch/<arch>/src/<chip>`  contains mainly chips or, at least, core definitions of these chips. We kept `esp32c3`, `esp32c6` and `esp32h2` in this level because it refers to a chip. However, they share common sources and headers and, in order to avoid confusion, it seemed that the best place for each would be `nuttx/arch/risc-v/src/common/espressif` (just like `nuttx/arch/risc-v/src/common` contains common source code for all RISC-V, `nuttx/arch/risc-v/src/common/espressif` contains common source code for all Espressif's RISC-V chips). Does it make sense?

Reply:
if we look at the board layout which make more sense:
```
ls boards/risc-v/espressif/
common/          esp32c3-generic/ esp32c6-generic/ esp32h2-generic/
```